plugin/cscope_maps.vim

```diff
@@ -41,12 +41,40 @@ if has("cscope")
     " if you want the reverse search order.
     set csto=0
 
-    " add any cscope database in current directory
-    if filereadable("cscope.out")
-        cs add cscope.out
-    " else add the database pointed to by environment variable
-    elseif $CSCOPE_DB != ""
+    " cscope seems to not index files in symlinked directories.
+    " However, one can build the cscope.files (list of files for cscope to index)
+    " via a command like:
+    " `find -L [your searched folder] -name [your searched pattern]
+    "   -exec realpath {} \; >> cscope.files`
+    "
+    " Note that it is prudent to use absolute file paths, hence the use of realpath.
+    "
+    " Now the trouble is getting vim to try to find a cscope.out file
+    " going recursively up through directories, but going back up through
+    " them via any symlinks you took to get to your CWD in the shell.
+    " The following codes solves this issue.
+
+    if $CSCOPE_DB != ""
         cs add $CSCOPE_DB
+    else
+        " Get all parts of out current path
+        let dirs = split($PWD, '/')
+        " Start building a lists of paths in which to look for cscope.out
+        let paths = ['/']
+        " /foo/bar/baz would result in the `paths` array containing:
+        " [/ /foo /foo/bar /foo/bar/baz]
+        for d in dirs
+            let paths = add(paths, paths[len(paths) - 1] . d . '/')
+        endfor
+
+        " List is backwords search order, so reverse it.
+        for d in reverse(paths)
```

Comment:
What about time penalty? Have you checked how vim startup time changes with and without this patch, for big project like Linux kernel? It would be a deal-breaker for me, if vim becomes "laggy" in Linux kernel. My usual day-to-day usage is opening and closing vim often, for Linux kernel project, and I expect vim to open instantly. Can you please check this for me and add your observations to commit message?

Reply:
I am also making use of these changes in Linux (specifically, symlinked-in sub directories :)).

I have not found any delay even when in this directory:
`/home/pdietl/git/starlab/lure/build/v4.19.56-1/tools/testing/selftests/rcutorture/formal/srcu-cbmc/tests/store_buffering`

Reply:
So you checked the case with Linux kernel indexing (not just tests), and there is no regression? If so, I'm fine with this.